src/CentrifugeRouter.sol

```diff
@@ -43,7 +48,8 @@ contract CentrifugeRouter is Auth, ICentrifugeRouter {
         uint256 lockedRequest = lockedRequests[user][vault];
         require(lockedRequest > 0, "CentrifugeRouter/user-has-no-balance");
         lockedRequests[user][vault] = 0;
-        IERC7540(vault).requestDeposit(lockedRequest, msg.sender, address(this));
+        IERC20(IERC7540(vault).asset()).approve(vault, lockedRequest);
+        IERC7540(vault).requestDeposit(lockedRequest, user, address(this));
```

Comment:
Likewise why is this reverted?

Reply:
This one needs to stay like this in order to let anybody call `executeLockedDepositRequest`

Reply:
Or integrate the above into `PoolManager.deployVault`, but that is pretty tight coupling

Reply:
Riiiight. Great idea! Multicall adds a lot of potential :)